kgforge/specializations/models/rdf/service.py

```diff
@@ -272,3 +350,191 @@ def traverse_properties(properties) -> Tuple[Dict, Dict]:
+        for sh_node in set(sh_nodes):
+            if str(sh_node) != str(node_shape_uriref) and str(sh_node) != str(RDF.nil):
+                t_a, p_a, t_r, c_r = self._get_property_shapes_from_nodeshape(
+                    sh_node, schema_graph
+                )
+                if p_a:
+                    triples_to_add.extend(t_a)
+                    rdfcollection_items_to_remove.extend(c_r)
+                    for propertyShape in p_a:
+                        triples_to_add.append(
+                            (node_shape_uriref, propertyShape[0], propertyShape[1])
+                        )
+                        property_shapes_to_add.append(propertyShape)
+                    triples_to_remove.extend(t_r)
+                    triples_to_remove.append((node_shape_uriref, SH.node, sh_node))
+                    for t in schema_graph.subjects(SH.node, sh_node):
+                        for list_, first, o, g in self._dataset_graph.quads(
+                            (
+                                None,
+                                URIRef(
+                                    "http://www.w3.org/1999/02/22-rdf-syntax-ns#first"
+                                ),
+                                t,
+                                None,
+                            )
+                        ):
+                            if g == self._get_named_graph_from_shape(node_shape_uriref):
+                                list_collection = RDFCollection(schema_graph, list_)
+                                t_index = list_collection.index(t)
+                                # del list_collection[t_index]
+                                rdfcollection_items_to_remove.append(
+                                    (list_collection, t_index)
+                                )
+        for sh_node_property in set(sh_properties):
+            if str(sh_node_property) != str(node_shape_uriref) and str(
+                sh_node_property
+            ) != str(RDF.nil):
+                property_shapes_to_add.append((SH.property, sh_node_property))
+        return (
+            triples_to_add,
+            property_shapes_to_add,
+            triples_to_remove,
+            rdfcollection_items_to_remove,
+        )
+
+    def get_shape_graph(
+        self,
+        node_shape_uriref: URIRef,
+        link_property_shapes_from_ancestors: bool = False,
+    ) -> Tuple[Shape, Graph]:
+        try:
+            shape = self.get_shape_graph_wrapper().lookup_shape_from_node(
+                node_shape_uriref
+            )
+            if (
+                node_shape_uriref in self.shape_to_defining_resource
+                and self.shape_to_defining_resource[node_shape_uriref] in self._imported
+            ):
+                shape_graph = self._dataset_graph.graph(
+                    self._get_named_graph_from_shape(node_shape_uriref)
+                )
+            else:
+                raise ValueError()
+        except Exception:
```

Comment:
This is quite broad, what can cause this? 

Can you explain what goes on in terms of logic, what sequence of events will go through the try, and what sequence of events will go through the except?